corehq/apps/data_dictionary/static/data_dictionary/js/data_dictionary.js

```diff
@@ -59,6 +59,35 @@ hqDefine("data_dictionary/js/data_dictionary", [
             self.deprecated(false);
         };
 
+        self.showAllowedValues = function () {
```

Comment:
The "Edit Group" page is a good example of this key/value widget. The js logic for creating the widget is [here](https://github.com/dimagi/commcare-hq/blob/af379325384d4383c6d4e49fbf2bb59cabe8f9a6/corehq/apps/groups/static/groups/js/group_members.js#L17-L23). The HTML for the editor is just [this](group-data-ui-editor) empty div, while a read-only version of the data is [here](https://github.com/dimagi/commcare-hq/blob/af379325384d4383c6d4e49fbf2bb59cabe8f9a6/corehq/apps/groups/templates/groups/group_members.html#L183-L198).

Reply:
The custom data fields profiles setup might also be a useful example, because it uses the same widget in the context of a more complex page that uses knockout. Widget setup is [here](https://github.com/dimagi/commcare-hq/blob/af379325384d4383c6d4e49fbf2bb59cabe8f9a6/corehq/apps/custom_data_fields/static/custom_data_fields/js/custom_data_fields.js#L110) and HTML is [here](https://github.com/dimagi/commcare-hq/blob/af379325384d4383c6d4e49fbf2bb59cabe8f9a6/corehq/apps/custom_data_fields/templates/custom_data_fields/partials/profiles_tab.html#L28).

Reply:
awesome, thank you.